### PR TITLE
Fix error status code and written unittests for Resend Email

### PR DESCRIPTION
**Pull request by harshakhmk**
### Description

Fixed status code, error responses for Resend Email endpoint and written unit tests

Fixes # [625](https://github.com/anitab-org/mentorship-backend/issues/625)

### Type of Change:

- Code
- Quality Assurance
- User Interface
- Outreach
- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder
- [x] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by harshakhmk** [comment]
@isabelcosta  @vj-codes @devkapilbansal I have raised an MR kindly review it, let me know if there any changes

File: tests/users/test_api_resend_email.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+from app.database.models.user import UserModel
+from app.database.sqlalchemy_extension import db
+from tests.base_test_case import BaseTestCase
+from tests.test_data import user1
+
+
+class TestResendEmailApi(BaseTestCase):
+    def setUp(self):
+        super(TestResendEmailApi, self).setUp()
```
Comment (devkapilbansal) [inline tests/users/test_api_resend_email.py]: Why this is required?

File: tests/users/test_api_resend_email.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        self.verified_user = UserModel(
+            name=user1["name"] + "Example",
```
Comment (devkapilbansal) [inline tests/users/test_api_resend_email.py]:
```suggestion
            name=user1["name"],
```

File: tests/users/test_api_resend_email.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            email=user1["email"],
+            username=user1["username"],
+            password=user1["password"],
+            terms_and_conditions_checked=user1["terms_and_conditions_checked"],
+        )
```
Comment (devkapilbansal) [inline tests/users/test_api_resend_email.py]: If I am not wrong, dictionary unpacking should work
```suggestion
        self.verified_user = UserModel(**user1)
```

File: tests/users/test_api_resend_email.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    # This function checks if the status code returns 409 for exisiting email
+    def test_error_code_for_resend_email(self):
+        test_payload = {"email": self.verified_user.email}
+        actual_response = self.client.post(
+            "user/resend_email",
+            follow_redirects=True,
+            content_type="application/json",
+            data=json.dumps(test_payload),
+        )
+        self.assertEqual(409, actual_response.status_code)
```
Comment (devkapilbansal) [inline tests/users/test_api_resend_email.py]: Don't use hard coded values
```suggestion
        self.assertEqual(HTTPStatus.CONFLICT, actual_response.status_code)
```

File: tests/users/test_api_resend_email.py (hunk 3 above)
Comment (devkapilbansal) [reply tests/users/test_api_resend_email.py]: If that works fine, we will open an issue to refactor other places too.

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Hi @harshakhmk I have some minor requests. Please look into the same

**Comment by harshakhmk** [comment]
@devkapilbansal  updated the code

File: tests/users/test_api_resend_email.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+        self.verified_user = UserModel(**user1)
+
+        self.verified_user.is_email_verified = True
+        self.verified_user.is_available = False
+        db.session.add(self.verified_user)
+        db.session.commit()
+
+    # This function checks if the status code returns 409 for exisiting email
+    def test_error_code_for_resend_email(self):
```
Comment (devkapilbansal) [inline tests/users/test_api_resend_email.py]: Use descriptive names
```suggestion
    def test_resend_email_user_already_verified(self):
```

File: tests/users/test_api_resend_email.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        test_payload = {"email": self.verified_user.email}
+        actual_response = self.client.post(
+            "user/resend_email",
+            follow_redirects=True,
+            content_type="application/json",
+            data=json.dumps(test_payload),
+        )
+        self.assertEqual(HTTP_STATUS.CONFLICT, actual_response.status_code)
```
Comment (devkapilbansal) [inline tests/users/test_api_resend_email.py:31]: Please assert the message returned too here

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Great work @harshakhmk 
Just some minor changes required :point_down:

**Comment by harshakhmk** [comment]
I have updated the commit @devkapilbansal @isabelcosta

**Review by devkapilbansal — APPROVED** [review]
LGTM :+1:

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@harshakhmk can you please add some more test cases too here?
I just found that Resend Email wasn't tested before. One case that can be tested is resend mail successfully.

@isabelcosta please approve the workflow run as @harshakhmk is first time contributor.

**Comment by isabelcosta** [comment]
@devkapilbansal thank you for pinging me for this 🙏🏾 
And great work reviewing PR!

**Comment by devkapilbansal** [comment]
@harshakhmk tests are failing. Please fix them and update the code.

**Comment by vj-codes** [comment]
@harshakhmk any updates about this PR?

**Pull request closed without merging**